### PR TITLE
Initial search information

### LINKED ISSUE #13: User Enters Initial Search Information
As a Renter, when I visit the Search page, I want to see a field to enter a location (zip code, address) and checkboxes to select the booking purpose, so that I can click a button “Yard Me” to view the yards that are available for the specified purposes. Then I am redirected to the yard Search Results page.

### PR DESCRIPTION
**Pull request by domo2192**
Things that happened 
-Create index action to bring us to the initial search page 
-Create form that requests location and purposes for search 
-Create find yards action in search controller for our form to route to 
-Moved that sets purposes to application controller and memoized 

Testing 
-No sad paths are accounted for or tested yet

### CODE REVIEW

File: spec/features/host/dashboard/index_spec.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@
     end
   end
 
-
+    #####
```
Comment (abreaux26) [inline spec/features/host/dashboard/index_spec.rb]: Do you mind removing this please?

File: spec/features/host/yards/new_spec.rb [hunk 2]
```diff
@@ -48,7 +48,7 @@
       expect(page).to have_button('Create Yard')
     end
   end
-
+  #######
```
Comment (abreaux26) [inline spec/features/host/yards/new_spec.rb]: Can you remove this as well?

File: spec/features/yards/show_spec.rb [hunk 3]
```diff
@@ -64,7 +64,7 @@
       expect(page).to have_button('Edit Yard')
     end
   end
-
+  #######
```
Comment (abreaux26) [inline spec/features/yards/show_spec.rb]: and this?

**Review by abreaux26 — CHANGES_REQUESTED** [review]
Nice work! I think you and Alexa might have merge conflicts?

**Comment by domo2192** [comment]
> Nice work! I think you and Alexa might have merge conflicts?

I think Alexas is failing travis because of changes main has that her local doesn't

**Review by abreaux26 — APPROVED** [review]

**Review by dev-named-jordan — APPROVED** [review]
Awesome, thanks for the changes!

**Pull request merged**